packages/strapi-design-system/src/IconButton/IconButton.js

```diff
@@ -126,10 +140,22 @@ IconButton.defaultProps = {
   disabled: false,
   onClick: undefined,
 };
+
+/**
+ * @type {(otherProps: string[]) => (props: Record<string, unknown>, propName: string) => Error | undefined}
+ */
+const throwPropErrorIfNoneAreDefined = (otherProps) => (props, propName) => {
+  if (!props[propName] && otherProps.every((otherProp) => !props[otherProp])) {
+    return new Error(`One of the following props is required: ${propName}, ${otherProps.join(', ')}`);
+  }
+};
+
 IconButton.propTypes = {
+  ['aria-label']: throwPropErrorIfNoneAreDefined(['label']),
+  children: throwPropErrorIfNoneAreDefined(['icon']),
   disabled: PropTypes.bool,
-  icon: PropTypes.element.isRequired,
-  label: PropTypes.string,
+  icon: throwPropErrorIfNoneAreDefined(['children']),
+  label: throwPropErrorIfNoneAreDefined(['aria-label']),
```

Comment:
These helpers now show up in the user-facing documentation. Could you overwrite these with some better documentation (using code-comments)?

<img width="1043" alt="Screenshot 2022-11-02 at 15 07 08" src="https://user-images.githubusercontent.com/2244375
/199510589-3ce6caa5-7444-4b12-81ad-27824425ae51.png">

Reply:
I didn't know that was possible, do you have an example / doc, but yes, I agree this would be better off resolved.

Reply:
`BoxProps.js` contains examples. I didn't know either but storybook seems to parse code comments and use them to overwrite the defaults.

Reply:
nice, thats cool.